puppet_forge.gemspec

```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "faraday", "~> 2.0"
   spec.add_runtime_dependency "faraday-follow_redirects", "~> 0.3.0"
   spec.add_dependency "semantic_puppet", "~> 1.0"
-  spec.add_dependency "minitar", "< 1.0.0 "
+  spec.add_dependency "minitar", RUBY_VERSION >= '3.1.0' ? '~> 1.0.0' : '< 1.0.0'
```

Comment:
Does that work properly? Do you know any other gems that do it like this?

Reply:
validated with bundle install, this add the required dependency as per ruby version. Not sure if any other gem does it like this. If not this I believe we cannot support ruby < 3.1

Reply:
het @bastelfreak are we good with this?

Reply:
I don't know if this works for gems published to rubygems.org. Personally I would drop Ruby 3.1 and older because it's EoL and just support Ruby 3.2.

Reply:
Okay,I will discuss about dropping the support with team, thanks you...